### PR TITLE
Scaling of cluster nodes

### PR DESCRIPTION
**Pull request by atengler**
This PR contains following changes:

- Fix compatibility with new version of Azure Contrainer Service API. Raise version of `azure-mgmt-containerservice` package to 3.0.0 and implement new way of getting Kubeconfig
- Add `resize` method to base engine class to allow cluster node scaling
- Add `Resizing` state for Cluster object
- Implement resize for GCE and AKS engines
- Add `cluster_resize` API endpoint

### CODE REVIEW

File: kqueen/blueprints/api/views.py [hunk 1]
```diff
@@ -179,6 +179,26 @@ def cluster_progress(pk):
...
+@api.route('/clusters/<uuid:pk>/resize', methods=['PATCH'])
+@jwt_required()
+def cluster_resize(pk):
+    obj = get_object(Cluster, pk, current_identity)
+
+    data = request.json
+    if not isinstance(data, dict) or (isinstance(data, dict) and 'node_count' not in data):
+        abort(400)
+
+    res_status, res_msg = obj.engine.resize(data['node_count'])
```
Comment (tomkukral) [inline kqueen/blueprints/api/views.py]: I'd prefer to have `resize` method on cluster instance.

File: kqueen/engines/aks.py [hunk 2]
```diff
@@ -125,47 +160,17 @@ def provision(self, **kwargs):
...
-                            }
-                        ]
-                    }
-                }
-            }
-        }
+        managed_cluster = ManagedCluster(
+            self.location,
+            dns_prefix=self.resource_group_name,
+            kubernetes_version='1.7.7',
```
Comment (tomkukral) [inline kqueen/engines/aks.py]: Please keep `TODO: fix hardcoded params`

File: kqueen/blueprints/api/views.py (hunk 1 above)
Comment (atengler) [reply kqueen/blueprints/api/views.py]: We currently do it this way on all views, so maybe we can move it at some later date during refactoring

File: kqueen/engines/aks.py (hunk 2 above)
Comment (atengler) [reply kqueen/engines/aks.py]: OK, will do

**Pull request merged**